src/hotspot/cpu/x86/x86.ad

```diff
@@ -3683,6 +3701,18 @@ instruct convHF2F_reg_reg(regF dst, rRegI src) %{
   ins_pipe( pipe_slow );
 %}
 
+instruct vconvHF2F(vec dst, vec src) %{
+  match(Set dst (HF2FV src));
+  format %{ "vector_conv_HF2F $dst,$src" %}
+  ins_encode %{
+    int src_size = Matcher::vector_length_in_bytes(this, $src);
+    int dst_size = src_size * 2;
+    int vlen_enc = vector_length_encoding(dst_size);
```

Comment:
This could now be changed to:
int vlen_enc = Matcher::vector_length_encoding(this);